bodhi/server/models.py

```diff
@@ -2543,8 +2543,12 @@ def test_cases(self):
     def full_test_cases(self):
         tests = set()
         for build in self.builds:
-            for test in build.package.test_cases:
-                tests.add(test)
+            test_cases = build.package.test_cases
+            unique_tests = dict()
```

Comment:
This data structure could be a ```set``` instead of a ```dict```. Since the values stored in the dict aren't important, it seems like you are mostly trying to keep a uniqueness on the keys.